### ISSUE TITLE
FastCCI prod server down

### ISSUE THREAD

**Issue by ujjwalagrawal17** [Commonists/fastcci#6]
**Summary**
The FastCCI server seems to be down. 
Link: https://fastcci1.wmflabs.org/
Message: 502 Bad Gateway.

**Comment by Jeff-Gdot** [comment]
The status page at https://fastcci1.wmflabs.org/status is timing out with error 502, and attempts to use FastCCI from a category page just hang with 'Connecting...'. I didn't see any notices about this being an issue, and it's been down for at least a day. https://commons.wikimedia.org/wiki/Commons:Help_desk#FastCCI_appears_to_be_down.

**Comment by dschwen** [comment]
Yeah, I've been on the server during an "outage". The server code is running but does not reply to HTTP requests. I'm investigating this now.

**Comment by dschwen** [comment]
Ok, I rebuild the servers a few hours ago. Let's see. So far so god.

**Comment by ujjwalagrawal17** [comment]
Thanks @dschwen

**Comment by ujjwalagrawal17** [comment]
Umm, Can you tell where can we find API documentation for FastCCI. I am a GSoC Intern working on Wikimedia Commons app. We were planning to integrate it into the app. 
Related Issue:- https://github.com/commons-app/apps-android-commons/issues/1221

**Comment by ujjwalagrawal17** [comment]
@dschwen It would be nice if we will recieve a paginated response instead of all the images. The readme has some API documentation but I think according to that we will recieve all images at once. Is the feature already available ??

**Comment by dschwen** [comment]
Ujjwal, the query and response formats are documented in the README.md file in the repo. If you want me to clarify some details further please open another ticket. The response you get is a bunch of page IDs. It is a very Compact amount of data. You can do the pagination on the client side (that's what I do). You need Commons API calls to fetch the image URLs.

**Comment by jonadem** [comment]
Seems to be down at the moment.
